commit/merkleroot/rmn/client.go

```diff
@@ -19,13 +41,595 @@ type Client interface {
 	ComputeReportSignatures(
 		ctx context.Context,
 		destChain *rmnpb.LaneDest,
-		requestedUpdates []rmnpb.FixedDestLaneUpdateRequest,
+		requestedUpdates []*rmnpb.FixedDestLaneUpdateRequest,
 	) (*ReportSignatures, error)
 }
 
 type ReportSignatures struct {
-	// ReportSignatures are the ECDSA signatures for the lane updates for each node.
+	// Signatures are the ECDSA signatures for the lane updates for each node.
+	Signatures []*rmnpb.EcdsaSignature
+	// LaneUpdates are the lane updates for each source chain that we got the signatures for.
 	// NOTE: A signature[i] corresponds to the whole LaneUpdates slice and NOT LaneUpdates[i].
-	Signatures  []rmnpb.EcdsaSignature
-	LaneUpdates []rmnpb.FixedDestLaneUpdate
+	LaneUpdates []*rmnpb.FixedDestLaneUpdate
+}
+
+// client is the base RMN Client implementation.
+type client struct {
+	lggr                                    logger.Logger
+	rawRmnClient                            RawRmnClient
+	rmnNodes                                []RMNNodeInfo
+	rmnRemoteAddress                        cciptypes.Bytes
+	rmnHomeConfigDigest                     cciptypes.Bytes
```

Comment:
I believe would always be a `Bytes32`, but Kostis can confirm.

```suggestion
	rmnHomeConfigDigest                     cciptypes.Bytes32
```

Reply:
Reason I didn't use `cciptypes.Bytes32` was to be aligned with proto generated types, so I don't have to convert each time I want to use it.